R/zchunk_LB142.ag_Fert_IO_R_C_Y_GLU.R

```diff
@@ -1,103 +1,227 @@
+      mutate(value = value * adj,                                        # Adjust production
+             value = value * CONV_T_MT) %>%                              # Convert unit from ton to million ton
+      select(-adj) ->
+      L142.ag_Fert_Prod_MtN_ctry_Y
+
+    # Aggregate N fertilizer productionto GCAM region level
+    L142.ag_Fert_Prod_MtN_ctry_Y %>%
+      left_join_error_no_match(iso_GCAM_regID, by = "iso") %>%           # Match in GCAM region ID
+      group_by(GCAM_region_ID, year) %>%
+      summarise(value = sum(value)) %>%                                  # Aggregate to region total
+      ungroup() %>%                                                      # Ungroup before complete
+      mutate(GCAM_commodity = aglu.FERT_NAME) %>%                             # Add GCAM commodity category for N fertilizer
+      complete(GCAM_region_ID = unique(iso_GCAM_regID$GCAM_region_ID),
+               GCAM_commodity, year, fill = list(value = 0)) ->          # Fill in missing region with 0
+      L142.ag_Fert_Prod_MtN_R_Y
+
+    # Aggregate N fertilizer consumption to GCAM region level
+    L100.FAO_Fert_Cons_tN %>%
+      select(iso, year, value) %>%
+      left_join_error_no_match(iso_GCAM_regID, by = "iso") %>%           # Match in GCAM region ID
+      group_by(GCAM_region_ID, year) %>%
+      summarise(value = sum(value),                                      # Aggregate to region total
+                value = value * CONV_T_MT) %>%                           # Convert unit from ton to million ton
+      ungroup() %>%                                                      # Ungroup before complete
+      mutate(GCAM_commodity = aglu.FERT_NAME) %>%                             # Add GCAM commodity category for N fertilizer
+      complete(GCAM_region_ID = unique(iso_GCAM_regID$GCAM_region_ID),
+               GCAM_commodity, year, fill = list(value = 0)) ->          # Fill in missing region with 0
+      L142.ag_Fert_Cons_MtN_R_Y
+
+    # Calculate net exports of N fertilizer
+    L142.ag_Fert_Prod_MtN_R_Y %>%
+      rename(prod = value) %>%
+      # Match in fertilizer consumption
+      left_join_error_no_match(L142.ag_Fert_Cons_MtN_R_Y, by = c("GCAM_region_ID", "GCAM_commodity", "year")) %>%
+      # Calculate net exports as production minus consumption
+      mutate(value = prod - value) %>%
+      select(-prod) ->
+      L142.ag_Fert_NetExp_MtN_R_Y
+
+    # Calculate fertilizer input-output coefficients, scaled so that consumption of fertilizer balance
+    # First, downscale fertilizer demands by country and crop to GLU
+    # NOTE: Allocate fertilizer consumption to GLUs on the basis of production, not harvested area
+
+    # Calculate agricultural prodcution total by country and crop
+    L100.LDS_ag_prod_t %>%
+      group_by(iso, GTAP_crop) %>%
+      summarise(total = sum(value)) %>%
+      ungroup() -> L142.ag_Prod_t_ctry_crop
+
+    # Agricultural production by country / crop / GLU
```

Comment:
This may set a record for longest pipeline (maybe?). It could probably use a little more detailed explanation up top, here, of what's about to happen.